src/datachain/lib/signal_schema.py

```diff
@@ -117,40 +146,119 @@ def from_column_types(col_types: dict[str, Any]) -> "SignalSchema":
                 )
         return SignalSchema(signals)
 
-    def serialize(self) -> dict[str, str]:
-        signals = {}
+    @staticmethod
+    def serialize_custom_model_fields(fr: type, custom_types: dict[str, Any]) -> str:
+        """This serializes any custom type information to the provided custom_types
+        dict, and returns the name of the type provided."""
+        orig = get_origin(fr)
+        if orig in (Literal, LiteralEx):
+            # Literal has no __name__ in Python 3.9
+            name = "Literal"
+        else:
+            name = str(fr.__name__)
+        if (
+            not ModelStore.is_pydantic(fr)
+            or not issubclass(fr, BaseModel)
+            or name in ("File", "TextFile", "ImageFile")
```

Comment:
Will it be useful in future to have the list of these types somewhere as a constant?

Reply:
I agree, although I'm not sure where to put them. Where do you think would be best for a "Built-In Feature Types" list? (As in, easy to maintain and not miss that these are listed somewhere.)

Reply:
Is there much harm in dropping this condition and serializing custom fields even for those built-in types? That way, even if a future version were to drop or rename any of these, old datasets could still be read as expected.

Reply:
Upon further consideration, I decided to just serialize those built-in models as well, since the default is to use the existing and defined model in the code first, this should be harmless unless they are renamed or deleted in the future, in which case having them serialized would be useful for reading old datasets.